src/app/replicator/main.coffee

```diff
@@ -471,29 +471,35 @@ module.exports = class Replicator extends Backbone.Model
         options.live = false
         @startRealtime options, (err) =>
             @set 'inSync', false
-            # Skip first synchronisation
-            # todo: find better solution
             if err?.status is 404
-                console.info 'The above 404 is not normal, but we retest the \
-                              synchronisation after.'
-                log.warn err
-                return callback()
+                console.info 'The above 404 is not normal, but we retest with \
+                              smallest pouch last_seq (remoteCheckpoint).'
+
+                @checkpointLoop ?= 0
+                @checkpointLoop++
+
+                if @checkpointLoop > 10
+                    return callback()
+                else
+                    options.remoteCheckpoint--
+                    return @sync options, callback
+
```

Comment:
You have a return in both branches (then and else). So the `callback err` below is never called